packages/cozy-mespapiers-lib/src/helpers/queries.js

```diff
@@ -68,7 +70,10 @@ export const buildFilesQueryByLabel = label => ({
     Q(FILES_DOCTYPE)
       .where({
         'metadata.qualification': {
-          label: label
+          label: {
+            $eq: label,
+            ...(flag('hide.healthTheme.enabled') && { $nin: healthLabels })
+          }
```

Comment:
The index should be on `metadata.qualification.label` rather than `metadata.qualification`, don't you have a warning on this query?
Also, keep in mind the `$nin` operator will force a full-scan of the whole index (which is somewhat acceptable as long as there are not too many qualified files).

Reply:
nouvelle approche : on ne touche plus aux requêtes